Wind bug fixes

lsiess opened the pull request:
Type of PR: 
Bug fixes

Description:
Fix bugs, clean code and improve comments

- fix issue #[391 mass loss rate in wind is not equal to the value specified in the input file](https://github.com/danieljprice/phantom/issues/391)
- fix critical error in the calculation of the dust temperature (ray-tracing routines)
- in wind_setup, if wind_temperature = 0, sets wind_temperature = Tstar
- improve comments in ray-tracing routines

Testing:
ran simulations with all ray-tracing options in both single and binary star systems

Did you run the bots? yes

danieljprice commented on src/main/inject_wind.F90:
```diff
@@ -469,10 +469,6 @@ subroutine inject_particles(time,dtlast,xyzh,vxyzu,xyzmh_ptmass,vxyz_ptmass,&
-       ! update the sink particle mass
-       if (nptmass > 0 .and. wind_emitting_sink <= nptmass) then
-          xyzmh_ptmass(4,wind_emitting_sink) = xyzmh_ptmass(4,wind_emitting_sink) - mass_of_spheres
-       endif
```
this is the only line I am concerned about, it means that we no longer conserve mass when emitting stellar winds, that is the mass loss is not taken from the sink particle mass. 

Can you explain why this has been removed?

danieljprice commented on src/main/deriv.F90:
```diff
@@ -195,22 +197,51 @@ subroutine derivs(icall,npart,nactive,xyzh,vxyzu,fxyzu,fext,divcurlv,divcurlB,&
     call get_growth_rate(npart,xyzh,vxyzu,dustgasprop,VrelVf,dustprop,ddustprop(1,:))!--we only get ds/dt (i.e 1st dimension of ddustprop)
  endif
 
```
just a general comment here, I think it would be good to move the various options here into a daughter routine called get_dust_temperature () where all the various options are handled inside, rather than having them clutter the derivs routine.

I'm ok to merge this but would be nice to clean it up sometime soon..

danieljprice reviewed (commented):
looks good. Please can you explain why the mass loss accounting from the sink particle was removed?

Also what was the reason for #391 as to why the Mdot was off by a factor of two?

danieljprice commented:
Hi Lionel & Mats,

 I fixed the build failure. I cannot update this pull request directly, so instead I submitted the changes to your fork via a pull request, which when accepted should automatically update (and fix) this one. The only change is to get rid of the compiler warnings, plus I merged the latest code and fixed the conflict in the AUTHORS file.
 
Cheers,

Daniel

lsiess commented:
Hi Daniel

I just merge lsiess-master with my master branch and push it. 

Thanks for your help with the compiler option!

Cheers
Lionel

On Wed, 05 Apr 2023 18:22:31 -0700
Daniel Price ***@***.***> wrote:

> Hi Lionel & Mats,
> 
>  I fixed the build failure. I cannot update this pull request directly, so instead I
> submitted the changes to your fork via a pull request, which when accepted should
> automatically update (and fix) this one. The only change is to get rid of the compiler
> warnings, plus I merged the latest code and fixed the conflict in the AUTHORS file.
> Cheers,
> 
> Daniel
>
